File: 9.palindrome-number/python/src/palindrome_number.py
class PalindromeNumber:
  def do(self, num: int) -> bool:
    if self.isNegative(num) or self.isZeroEnding(num):
      return False

    reverted = 0

    while num > reverted:
      reverted = (
        self.carry_by_1_digit(reverted) +
        self.get_last_digit(num)
      )
      num = self.delete_last_digit(num)

    return (num == reverted) or (num == reverted // 10)

  def isNegative(self, num: int) -> bool:
    return num < 0

  def isZeroEnding(self, num: int) -> bool:
    return (num % 10 == 0)

  def carry_by_1_digit(self, reverted: int) -> int:
    return reverted * 10

  def get_last_digit(self, num: int) -> int:
    return num % 10

  def delete_last_digit(self, num: int) -> int:
    return num // 10

File: 9.palindrome-number/nodejs/src/palindrome_number.ts
class PalindromeNumber {
  public do(num: number): boolean {
    if (this.isNegative(num) || this.isZeroEnding(num)) {
      return false
    }

    let convertedNumber = 0

    while (num > convertedNumber) {
      convertedNumber = (this.carryBy1Digit(convertedNumber)
                        + this.getLastDigit(num))
      num = this.deleteLastDigit(num)
    }

    return (num == convertedNumber) || (num == Math.floor(convertedNumber / 10))
  }

  private isNegative(num: number): boolean {
    return num < 0
  }

  private isZeroEnding(num: number): boolean {
    return num % 10 === 0
  }

  private carryBy1Digit(convertedNumber: number): number {
    return convertedNumber * 10
  }

  private getLastDigit(num: number): number {
    return num % 10
  }

  private deleteLastDigit(num: number): number {
    return Math.floor(num / 10)
  }
}

export { PalindromeNumber }
